src/main/java/com/artipie/asto/s3/S3Storage.java

```diff
@@ -386,69 +282,49 @@ public S3Storage(final S3AsyncClient client, final String bucket, final boolean
     }
 
     /**
-     * Uploads content using multipart upload.
-     *
-     * @param key Object key.
-     * @param content Object content to be uploaded.
-     * @return Completion stage which is completed when upload is completed or aborted.
-     */
-    private CompletableFuture<Void> uploadMultipart(final Key key, final Content content) {
-        return this.client.createMultipartUpload(
-            CreateMultipartUploadRequest.builder()
-                .bucket(this.bucket)
-                .key(key.string())
-                .build()
-        ).thenApply(
-            created -> new MultipartUpload(
-                new Bucket(this.client, this.bucket),
-                key,
-                created.uploadId()
-            )
-        ).thenCompose(
-            upload -> upload.upload(content).handle(
-                (ignored, throwable) -> {
-                    final CompletionStage<Void> finished;
-                    if (throwable == null) {
-                        finished = upload.complete();
-                    } else {
-                        final CompletableFuture<Void> promise = new CompletableFuture<>();
-                        finished = promise;
-                        upload.abort().whenComplete(
-                            (result, ex) -> promise.completeExceptionally(throwable)
-                        );
-                    }
-                    return finished;
-                }
-            ).thenCompose(self -> self)
-        );
-    }
-
-    /**
-     * Translate {@link NoSuchKeyException} to {@link ValueNotFoundException} in future.
+     * Save multipart.
      *
-     * @param key Key that was accessed by the future.
-     * @param future Future to translate excpetion in.
-     * @param <T> Future result type.
-     * @return Translated future.
+     * @param updated The estimated content.
+     * @param key The key of value to be saved.
+     * @return The future.
      */
-    private static <T> CompletableFuture<T> handleNoSuchKey(
-        final Key key,
-        final CompletableFuture<T> future
-    ) {
-        return future.handle(
-            (content, throwable) -> {
-                CompletionStage<T> result = future;
-                if (throwable instanceof CompletionException) {
-                    final Throwable cause = throwable.getCause();
-                    if (cause instanceof NoSuchKeyException) {
-                        result = new FailedCompletionStage<>(
-                            new ValueNotFoundException(key, cause)
-                        );
+    private CompletableFuture<Void> putMultipart(final Content updated, final Key key) {
+        final CompletableFuture<Void> future;
+        final Optional<Long> size = updated.size();
+        if (size.isPresent() && size.get() < S3Storage.MIN_MULTIPART) {
+            future = this.put(key, updated);
```

Comment:
@andreoss it's a bit confusing: the method is named `putMultipart` but here we're using non-multipart simple upload

Reply:
@g4s8 Done